client/shared/src/util/useMatchMedia.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * Returns provided media query match and, by default, subscribes to its updates.
+ *
+ * Example: `const isSmallScreen = useMatchMedia('(max-width: 250px)')`
+ *
+ * @param query CSS media query to match
+ * @param observe Boolean flag indicating if you want to subscribe to updates, `true` by default
+ * @returns `boolean`
+ */
+export function useMatchMedia(query: string, observe = true): boolean {
+    const [isMatch, setIsMatch] = useState(window.matchMedia(query).matches)
+
+    useEffect(() => {
+        const handler = (event: MediaQueryListEvent): void => setIsMatch(event.matches)
+        if (observe) {
+            window.matchMedia(query).addEventListener('change', handler)
+        }
+        return () => {
+            if (observe) {
+                window.matchMedia(query).removeEventListener('change', handler)
+            }
```

Comment:
```suggestion
                window.matchMedia(query).removeEventListener('change', handler)
```

`observe` could still have been true previously, so it's safer just to always `removeEventListener` - which shouldn't error I think?